### DISCUSSION TITLE
Visual Studio build: No sound for MUS files

### DISCUSSION THREAD

**Discussion by stuerp** [libsidplayfp/libsidplayfp#56] [General]
Hi,

I took over development for [foo_sid](https://github.com/stuerp/foo_sid), a component for foobar2000. I wrestled with getting libsidplayfp to build on VS and got it working for SID files.

However, MUS and MUS+STR files simply give a 'click' at the beginning and go silent after that. Same behavior in 32 or 64-bit build; same behavior for resid and residfp.

Since I don't understand the libsidplayfp code, any ideas what is going on? sidplayfp 2.4.0 w64 plays the test file just fine so it must be the compiler settings.

**Comment by mywave82** [comment]
In general two thoughts comes to mind:
Do you have the c64 assembler code converter in ROM stub code as expected.
Some SID files etc do require kernal, chargen and/or basic rom files to be loaded.

**Reply by stuerp** [reply]
The kernal, chargen and basic roms were part of the original source. sidplayer1 and 2 were regenerated with xa but they seem wrong as @drfiemost suggests.

**Comment by drfiemost** [comment]
Your sidplayer bin files look bogus, you need [xa](http://www.floodgap.com/retrotech/xa/) to regenerate them or you can find'em in the libsidplayfp source tarball.
Tried a few MUS files out of curiosity and they play fine without any ROM loaded.

**Reply by stuerp** [reply]
I think you mean the sidplayer1 and-2 files? They were assembled with xa with everything set to default. Obviously I accepted the output without reservation. My 6502 assembler is a bit rusty.
Thanks for pointing me in a new direction. I'll try the tarball.

**Reply by drfiemost** [reply]
My bad, it also requires od from coreutils, the full recipe is in the makefile. BTW you'll be fine with the distributed preassembled files.

**Reply by stuerp** [reply]
Replacing the sidplayer files did the trick. Thanks a lot.
